### PR TITLE
Add tests for <map-span>

### PR DESCRIPTION
**Pull request by ben-lu-uw**
Closes https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/536

### CODE REVIEW

**Comment by ben-lu-uw** [comment]
Also closes https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/559

**Comment by prushforth** [comment]
Seems like the volume of data included in the commit exceeds what might be strictly necessary to test the function?

**Comment by ben-lu-uw** [comment]
I'll strip it down 👍

File: src/mapml/features/feature.js [hunk 1]
```diff
@@ -240,6 +240,11 @@ export var Feature = L.Path.extend({
     this._outline = [];
     for (let coords of this._markup.querySelectorAll('map-coordinates')) {
       let nodes = coords.childNodes, cur = 0, tempDiv = document.createElement('div'), nodeLength = nodes.length;
+      for(let i = 0; i < nodes.length; i++){
+        if(nodes[i].textContent.trim().length === 0){
+          nodes[i].remove();
+        }
+      }
```
Comment (ahmadayubi) [inline src/mapml/features/feature.js:247]: Do you need to loop through all to childNodes or could you get away with just checking just the first and last?

Comment (ben-lu-uw) [reply src/mapml/features/feature.js:247]: It seems that I only need to check the last one

File: test/e2e/core/mapSpan.test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                        await page.waitForNavigation();
+                    }
+                    await page.goto(PATH + "mapSpan.html");
+                });
+                afterAll(async function () {
+                    await browser.close();
+                });
+
+                test("[" + browserType + "]" + " <map-span> hides tile boundaries", async ()=>{
+                    const total = await page.$eval(
```
Comment (prushforth) [inline test/e2e/core/mapSpan.test.js:25]: Why is the variable named "total" ?  What is it the total of?

File: test/e2e/core/mapSpan.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    const spliced = await hidden.slice(3, hidden.length);
+                    //Makes sure that the part that should be hidden is not part of the feature outline
+                    let index = d.indexOf(spliced);
+
+                    expect(total).toEqual("stroke: none;");
+                    expect(index).toEqual(-1);
+                });
+
+                //https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/559#issuecomment-959805896
+                test("[" + browserType + "]" + " White space parsing for map-coordinates", async ()=>{
```
Comment (prushforth) [inline test/e2e/core/mapSpan.test.js:50]: Title of the test could be easier to understand, e.g. "Ensure white space in map-coordinates is ignored" or something like that.

File: test/e2e/core/mapSpan.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                    await page.waitForTimeout(1000);
+                    const feature = await page.$eval(
+                        'body > mapml-viewer:nth-child(2) div > div.leaflet-pane.leaflet-map-pane > div.leaflet-pane.leaflet-overlay-pane > div > div.leaflet-layer.mapml-templatedlayer-container > div > div > div:nth-child(1) > svg > g > g > path.fclass.mapml-feature-outline',
+                        (path) => path.getAttribute("d")
+                    );
+
+                    expect(feature).toEqual("M0 217L0 217L0 217L2 217L4 218L6 218L6 218L6 216L2 214L0 216L0 216");
```
Comment (prushforth) [inline test/e2e/core/mapSpan.test.js:57]: 👍

File: test/e2e/data/tiles/osmtile/2/1_1.mapml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        <map-meta charset="utf-8"/>
+        <map-meta content="text/mapml" http-equiv="Content-Type"/>
+      <map-meta name="projection" content="OSMTILE"></map-meta>
+      <map-meta name="cs" content="pcrs"></map-meta>
+    </map-head>
+    <map-body>
+        <map-feature id="fclass.73" class="fclass">
+            <map-geometry>
+                <map-polygon>
+                    <map-coordinates>
```
Comment (prushforth) [inline test/e2e/data/tiles/osmtile/2/1_1.mapml:13]: You could put an xml comment here that says e.g. `

**Review by prushforth — COMMENTED** [review]
Thanks for doing these test, it is much appreciated.  Testing is the basis of civilizational progress lol.  Just try to keep the semantics of what your doing up front for poor old you or whoever comes after you (often yourself, actually) who comes along and tries to figure it out later.

**Comment by prushforth** [comment]
If you feel like updating anything, go ahead, but merging this so you can move on as required.

**Pull request merged**